back/api/v1/authentication.go

```diff
@@ -0,0 +1,168 @@
+package v1
+
+import (
+	"4ctf/models"
+	"4ctf/utils"
+	"4ctf/views"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/savsgio/atreugo/v11"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+func setupAuthRoutes(api *Api, router *atreugo.Router) {
+	group := router.NewGroupPath("/auth")
+
+	group.POST("/login", WithDefaults(api, WithBody(login)))
+	group.POST("/register", WithDefaults(api, WithBody(register)))
+	group.GET("/logout", WithDefaults(api, logout))
+	group.GET("/me", WithDefaults(api, WithValidSession(profile)))
+}
+
+type LoginRequest struct {
+	Username string `json:"username" validate:"min=2,max=40,regexp=^[a-zA-Z]*$"`
+	Password string `json:"password" validate:"max=40,password"`
+}
+
+func login(api *Api) func(ctx *atreugo.RequestCtx, lr *LoginRequest) error {
```

Comment:
Could we also allow to user email to login instead of username or not ?

Reply:
We can, though it adds a bit of complexity, do we really want to ?

Reply:
username only  is also fine for me

Reply:
We can go on username only and keep the idea for later ?

Reply:
Yep, this can be added in a future PR, I suggest you open an issue to keep that in mind for later